src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2950,6 +2950,47 @@ private boolean isTemplateOf(JSType type, String typeName) {
           && type.toMaybeTemplatizedType().getReferenceName().equals(typeName);
     }
 
+    private boolean handleSpecialTypeAssertionFunctions(String propName) {
+      if (!isGoogNamespace) return false;
+      if (propName.equals("isArray")) {
+        emit("(val : any ) : val is any[] ");
+        return true;
+      }
+      if (propName.equals("isBoolean")) {
```

Comment:
I'm actively working on deprecating and (eventually) deleting all of these.  I'd rather not be encouraging their use.

Reply:
In particular, I've got an LSC in progress to replace `isString`, `isBoolean`, `isNumber`, `isDef`, `isNull`, and `isDefAndNotNull` for all internal JS and TS code in one fell swoop.

I may look into doing the other three as well, and possible `goog.typeof` if I can figure out how to do it safely.